uaclient/entitlements/tests/conftest.py

```diff
@@ -3,51 +3,68 @@
 from uaclient import config
 
 try:
-    from typing import Any, Dict, List  # noqa
+    from typing import Any, Dict, List, Optional  # noqa
 except ImportError:
     # typing isn't available on trusty, so ignore its absence
     pass
 
 
-def machine_token(entitlement_type: str) -> "Dict[str, Any]":
+def machine_token(
+    type: str,
+    *,
+    affordances: "Dict[str, Any]" = None,
+    directives: "Dict[str, Any]" = None,
+    entitled: "Optional[bool]" = True,
+    obligations: "Dict[str, Any]" = None,
+    suites: "List[str]" = None
+) -> "Dict[str, Any]":
     return {
+        "resourceTokens": [{"Type": type, "Token": "%s-token" % type}],
         "machineToken": "blah",
         "machineTokenInfo": {
             "contractInfo": {
                 "resourceEntitlements": [
-                    {"type": entitlement_type, "entitled": True}
+                    machine_access(
+                        type,
+                        affordances=affordances,
+                        directives=directives,
+                        entitled=entitled,
+                        obligations=obligations,
+                        suites=suites,
+                    )
                 ]
             }
         },
     }
 
 
 def machine_access(
-    entitlement_type: str,
+    type: str,
     *,
     affordances: "Dict[str, Any]" = None,
     directives: "Dict[str, Any]" = None,
+    entitled: "Optional[bool]" = True,
```

Comment:
I don't think this is `Optional`; it should only ever be `True` or `False`, so it's a straight-up `bool`. I just confirmed this for myself with a toy example:

```py
from typing import Optional


def optional_foo(x: Optional[bool] = True) -> None:
    pass


optional_foo(True)
optional_foo(False)
optional_foo()
optional_foo(None)


def foo(x: bool = True) -> None:
    pass


foo(True)
foo(False)
foo()
foo(None)
```

gives:

```
$ mypy foo.py
foo.py:21: error: Argument 1 to "foo" has incompatible type "None"; expected "bool"
```

Reply:
oopsie. got excited about the Optional. Dropped